### PR TITLE
Enumerate fewer cross-origin properties

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/html/pull/3186 for context.

### CODE REVIEW

**Comment by annevk** [comment]
So the one thing I don't get is that in https://bug1412741.bmoattachments.org/attachment.cgi?id=8924054 @bzbarsky also marked the [[OwnPropertyKeys]] test as having changed results. I don't see how that follows from the standard.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20109)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/296242747)

**Comment by bzbarsky** [comment]
> also marked the [[OwnPropertyKeys]] test as having changed results

The test in question is http://searchfox.org/mozilla-central/rev/423b2522c48e1d654e30ffc337164d677f934ec3/testing/web-platform/tests/html/browsers/origin/cross-origin-objects/cross-origin-objects.html#283-296 (or https://github.com/w3c/web-platform-tests/blob/26cc7b48a87800a0e28ea765e3630b1c929cf1c9/html/browsers/origin/cross-origin-objects/cross-origin-objects.html#L283-L296 if you prefer).  It tests the two obvious spec entrypoints that exercise `[[OwnPropertyKeys]]`: `Object.getOwnPropertyNames` and `Object.keys`.  The former is not affected by enumerability, but the latter is.

I didn't want to just change the test before the spec changes, so I just marked it as failing for now, but of course we'll want to update the test...

**Comment by annevk** [comment]
Thanks, I'll fix that one too. I suspect it would be better for @cdumez et al if we fixed it here straight away given that there's still some (significant) delay if we land web-platform-tests changes from the Firefox side.

**Comment by annevk** [comment]
Pushed a fix.

File: html/browsers/origin/cross-origin-objects/cross-origin-objects.html [hunk 1]
```diff
@@ -285,13 +287,12 @@
...
-                      whitelistedWindowPropNames,
+                      whitelistedWindowIndices,
                       "Object.keys() gives the right answer for cross-origin Window");
   assert_array_equals(Object.getOwnPropertyNames(C.location).sort(),
                       whitelistedLocationPropNames,
                       "Object.getOwnPropertyNames() gives the right answer for cross-origin Location");
-  assert_array_equals(Object.keys(C.location).sort(),
-                      whitelistedLocationPropNames,
+  assert_equals(Object.keys(C.location).length, 0,
                       "Object.keys() gives the right answer for cross-origin Location");
```
Comment (bzbarsky) [inline html/browsers/origin/cross-origin-objects/cross-origin-objects.html]: Fix the indent here?

**Review by bzbarsky — APPROVED** [review]
Looks good, thank you!

**Pull request merged**